pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -473,19 +473,23 @@ func (r *ReconcileClusterDeployment) syncDeletedClusterDeployment(cd *hivev1.Clu
 
 	// Skips creation of uninstall job if PreserveOnDelete is true
 	if cd.Spec.PreserveOnDelete {
-		cdLog.Warn("skipping creation of uninstall job, due to PreserveOnDelete")
-		if controllerutils.HasFinalizer(cd, hivev1.FinalizerDeprovision) {
-			err = r.removeClusterDeploymentFinalizer(cd)
-			if err != nil {
-				cdLog.WithError(err).Error("error removing finalizer")
+		if cd.Status.Installed {
+			cdLog.Warn("skipping creation of uninstall job, due to PreserveOnDelete")
```

Comment:
This message is not completely correct now.

Reply:
It looks about right to me, here we are skipping the job because installed=true and preserve=true.

Maybe just "skipping creation of uninstall job for installed cluster due to PreserveOnDelete=true" for clarity.